src/synthesizer/dust/emission.py

```diff
@@ -100,21 +123,54 @@ class Blackbody(EmissionBase):
     A class to generate a blackbody emission spectrum.
     """
 
+    temperature: unyt_quantity
+    cmb_heating: bool
+    z: float
+
     @accepts(temperature=temperature_dim)
-    def __init__(self, temperature):
+    def __init__(
+        self,
+        temperature: unyt_quantity,
+        cmb_heating: bool = False,
+        z: float = 0,
+    ) -> None:
         """
         A function to generate a simple blackbody spectrum.
 
         Args:
             temperature (unyt_array)
                 The temperature of the dust.
 
+            cmb_heating (bool)
+                Option for adding heating by CMB
+
+            z (float)
+                Redshift of the galaxy
+
         """
+        # emmissivity of true blackbody is 1
+        emissivity = 1.0
+        _temperature: unyt_quantity
+        cmb_factor: float
+        if cmb_heating:
```

Comment:
Like here, what is this whole block doing, what is going on with the derivation in this block. Some more context would help.

Reply:
Also, correct me if I'm wrong but isn't this the same for all models? Could we not make a private method for this and attach it to the parent class to reduce boilerplate?